src/services/db/GitFileSystemService.ts

```diff
@@ -0,0 +1,277 @@
+import fs from "fs"
+
+import { errAsync, okAsync, ResultAsync } from "neverthrow"
+import { SimpleGit } from "simple-git"
+
+import { config } from "@config/config"
+
+import GitFileSystemError from "@errors/GitFileSystemError"
+
+import { ISOMER_GITHUB_ORG_NAME } from "@constants/constants"
+
+import type { GitDirectoryItem, GitFile } from "@root/types/gitfilesystem"
+
+/**
+ * Some notes:
+ * - Seems like getTree, updateTree and updateRepoState is always used together
+ */
+
+const EFS_VOL_PATH = config.get("aws.efs.volPath")
+const BRANCH_REF = config.get("github.branchRef")
+
+export default class GitFileSystemService {
+  private readonly git: SimpleGit
+
+  constructor(git: SimpleGit) {
+    this.git = git
+  }
+
+  private async isGitInitialized(
+    repoName: string
+  ): Promise<ResultAsync<boolean, GitFileSystemError>> {
```

Comment:
oof sorry should have been clearer, i meant 
`ResultAsync<boolean,GitFileSystemError>` rather than `Promise<ResultAsync<boolean,GitFileSystemError>>`. The moment we wrap it in a promise, we lose the benefits of guaranteeing type safety for errors thrown (`rejects` can only be either typed to `unknown` or `any`).  Happy to huddle if this does not make sense.

Reply:
Done in f9fb3b337cd02d66c2c05eb50f6e88b389196d9b. Thanks so much for guiding me on how to properly use neverthrow! 🌮